### PR TITLE
Introduce PHPStan

### LINKED ISSUE #189: Use PHPStan for Static Analysis
**Issue Overview**

Something like https://packagist.org/packages/szepeviktor/phpstan-wordpress or even just PHPStan directly with custom stubs will be very useful for finding issues in the codebase using static analysis.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Versions**

* Browser: 
* PHP: 
* WP-CLI: 
* GlotPress: 
* WordPress: 

**Additional context**

### PR DESCRIPTION
**Pull request by szepeviktor**
Closes #189

@swissspidy I've cleaned up problems up to Level 2.

Level 5 is desired.
Level max is dreamed.

### CODE REVIEW

**Comment by szepeviktor** [comment]
```php
wp_unschedule_event( int $timestamp, string $hook, array $args = array() )
```

File: inc/CLI/LanguagePackCommand.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ public function list( $args, $assoc_args ): void {
 				'Locale'       => $locale->wp_locale,
 				'English Name' => $locale->english_name,
 				'Native Name'  => $locale->native_name,
-				// Adding 0 removes trailing zeros.
+				// Adding 0 removes trailing zeros -> FIXME Use a PHP rounding function.
 				'Completed'    => sprintf( '%s%%', number_format( $set->percent_translated(), 1 ) + 0 ),
```
Comment (swissspidy) [inline inc/CLI/LanguagePackCommand.php:104]: Hacky 😅

File: inc/CLI/LanguagePackCommand.php [hunk 2]
```diff
@@ -100,7 +100,7 @@ public function list( $args, $assoc_args ): void {
 				'Locale'       => $locale->wp_locale,
 				'English Name' => $locale->english_name,
 				'Native Name'  => $locale->native_name,
-				// Adding 0 removes trailing zeros.
+				// Adding 0 removes trailing zeros -> FIXME Use a PHP rounding function.
```
Comment (szepeviktor) [inline inc/CLI/LanguagePackCommand.php:103]: or `%0.f` in sprintf

File: inc/CLI/LanguagePackCommand.php (hunk 1 above)
Comment (szepeviktor) [reply inc/CLI/LanguagePackCommand.php:104]: or %0.f in sprintf

File: composer.json [hunk 3]
```diff
@@ -38,8 +38,11 @@
   },
   "require-dev": {
     "dealerdirect/phpcodesniffer-composer-installer": "^0.7.0",
+    "php-stubs/wp-cli-stubs": "^2.4",
     "phpunit/phpunit": "^6.5 || ^7.5",
-    "wearerequired/coding-standards": "^1.5"
+    "szepeviktor/phpstan-wordpress": "^0.6.5",
+    "wearerequired/coding-standards": "^1.5",
+    "wpackagist-plugin/glotpress": "^2.3.1"
```
Comment (ocean90) [inline composer.json:45]: Can we please keep the plugin in the vendor directory instead of moving it into the default wp-content directory?

Comment (szepeviktor) [reply composer.json:45]: @ocean90 Done.

**Comment by ocean90** [comment]
With level 5 we still have these errors:

```
vendor/bin/phpstan analyze --memory-limit=-1
Note: Using configuration file traduttore/phpstan.neon.dist.
PHPStan crashed in the previous run probably because of excessive memory consumption.
It consumed around 61 MB of memory.


To avoid this issue, allow to use more memory with the --memory-limit option.
 32/32 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------
  Line   CLI/LanguagePackCommand.php
 ------ ----------------------------------------------------------------
  104    Binary operation "+" between string and 0 results in an error.
 ------ ----------------------------------------------------------------

 ------ --------------------------------------------
  Line   ProjectLocator.php
 ------ --------------------------------------------
  120    Ternary operator condition is always true.
  151    Ternary operator condition is always true.
 ------ --------------------------------------------

 ------ ----------------------------------------------------------------------------------------------------
  Line   Updater.php
 ------ ----------------------------------------------------------------------------------------------------
  71     Parameter #1 $timestamp of function wp_unschedule_event expects int, string given.
  71     Parameter #2 $hook of function wp_unschedule_event expects string, int<1, max>|int<min, -1> given.
 ------ ----------------------------------------------------------------------------------------------------

 ------ ----------------------------------------------------------------------------------------------------
  Line   ZipProvider.php
 ------ ----------------------------------------------------------------------------------------------------
  104    Parameter #1 $timestamp of function wp_unschedule_event expects int, string given.
  104    Parameter #2 $hook of function wp_unschedule_event expects string, int<1, max>|int<min, -1> given.
 ------ ----------------------------------------------------------------------------------------------------


 [ERROR] Found 7 errors
```

**Comment by szepeviktor** [comment]
Yes. I leave those up to you.

**Review by ocean90 — DISMISSED** [review]

**Comment by szepeviktor** [comment]
Should we add `"analyze": "vendor/bin/phpstan analyze"` to Composer scripts?

**Comment by szepeviktor** [comment]
Thank you.

**Pull request merged**